### DISCUSSION TITLE
Add Support for More Languages to the Xray API Project

### DISCUSSION THREAD

**Discussion by iamfarhad** [XTLS/Xray-core#3801] [Ideas]
Hello everyone,
Please take a look at this repository: https://github.com/XVGuardian/xray-api

This repository fetches all the proto files from the Xray project, stores them, and generates the related gRPC files for Python based on those proto files.
However, this project is currently limited to only a couple of languages.
Please create a similar project that supports other languages, such as PHP or Java Spring Boot, to make it easier for us to use it in our own projects.

**Comment by mmmray** [comment]
That's not the way to go about feature requests. If you want PHP and Java support (you already filed issues against the xray-api project it seems) and it doesn't get done, then it won't help to have an "official" project that is even more resource-constrained. I suggest to work with the author of xray-api to figure out what he needs to make your request happen.
